src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/ProjectPackages.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using NuGet.Packaging;
+
+namespace NuGet.PackageManagement.VisualStudio
+{
+    /// <summary>
+    /// Struct containing installed and transitive packages for a project
+    /// </summary>
+    public struct ProjectPackages
+    {
+        public IReadOnlyList<PackageReference> InstalledPackages { get; }
+        public IReadOnlyList<PackageReference> TransitivePackages { get; }
+
+        public ProjectPackages(IReadOnlyList<PackageReference> installedPackages, IReadOnlyList<PackageReference> transitivePackages) 
+        {
+            InstalledPackages = installedPackages;
+            TransitivePackages = transitivePackages;
+        }
+
+        public override bool Equals(object obj)
+        {
+            throw new NotImplementedException();
+        }
+
+        public override int GetHashCode()
+        {
+            throw new NotImplementedException();
+        }
+
+        public static bool operator ==(ProjectPackages left, ProjectPackages right)
+        {
+            return left.Equals(right);
```

Comment:
I assume that you throw above because you want to ensure this type is not used as a collection key, but I'm guessing.  You could have chosen to not override them at all, but you did.  So, it would be good to add a comment wherever you throw with the reason _why_ you chose to throw instead of not override or implement.

I'd rather you throw here and below explicitly rather than rely on `Equals(...)` throwing, which isn't obvious.  (I missed it in my previous reviews.)  That is, I assume you also meant for these operators to throw.  Is that correct?

These comments also apply [here](https://github.com/NuGet/NuGet.Client/pull/3680/files#diff-0b37cab887963fe6cb26878299731bb80f26dae8afc25028c907c65375889c1dR24-R42).

Reply:
I overrode these because Rule CA1815 requires it. I've changed them all to throw NotImlementedException. Or should I implement them even if I don't expect them to ever be used?

Reply:
going to resolve for now...and follow up with birgit and damon via email.